beacon-chain/rpc/aggregator/server_test.go

```diff
@@ -90,3 +99,132 @@ func TestSubmitAggregateAndProof_IsAggregator(t *testing.T) {
+	ctx := context.Background()
+
+	beaconState, privKeys := testutil.DeterministicGenesisState(t, 32)
+	att0 := GenerateAtt(beaconState, 0, privKeys)
+	att1 := GenerateAtt(beaconState, 1, privKeys)
+
+	beaconState.Slot += params.BeaconConfig().MinAttestationInclusionDelay
+
+	aggregatorServer := &Server{
+		HeadFetcher: &mock.ChainService{State: beaconState},
+		SyncChecker: &mockSync.Sync{IsSyncing: false},
+		BeaconDB:    db,
+		AttPool:     attestations.NewPool(),
+		P2p:         &mockp2p.MockBroadcaster{},
+	}
+
+	priv := bls.RandKey()
+	sig := priv.Sign([]byte{'B'}, 0)
+	pubKey := [48]byte{'B'}
+	req := &pb.AggregationRequest{CommitteeIndex: 1, SlotSignature: sig.Marshal(), PublicKey: pubKey[:]}
+	if err := aggregatorServer.BeaconDB.SaveValidatorIndex(ctx, pubKey, 100); err != nil {
+		t.Fatal(err)
+	}
+
+	if err := aggregatorServer.AttPool.SaveUnaggregatedAttestation(att0); err != nil {
+		t.Fatal(err)
+	}
+	if err := aggregatorServer.AttPool.SaveUnaggregatedAttestation(att1); err != nil {
+		t.Fatal(err)
+	}
+
+	if _, err := aggregatorServer.SubmitAggregateAndProof(ctx, req); err != nil {
+		t.Fatal(err)
+	}
+
+	aggregatedAtts := aggregatorServer.AttPool.AggregatedAttestation()
+	wanted, err := helpers.AggregateAttestation(att0, att1)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if reflect.DeepEqual(aggregatedAtts, wanted) {
+		t.Error("Did not receive wanted attestation")
+	}
+}
+
+func TestSubmitAggregateAndProof_AggregateNotOk(t *testing.T) {
+	params.UseMinimalConfig()
+	c := params.MinimalSpecConfig()
+	c.TargetAggregatorsPerCommittee = 16
+	params.OverrideBeaconConfig(c)
+	defer params.UseMainnetConfig()
+
+	db := dbutil.SetupDB(t)
+	defer dbutil.TeardownDB(t, db)
+	ctx := context.Background()
+
+	beaconState, privKeys := testutil.DeterministicGenesisState(t, 32)
+	att0 := GenerateAtt(beaconState, 0, privKeys)
+
+	beaconState.Slot += params.BeaconConfig().MinAttestationInclusionDelay
+
+	aggregatorServer := &Server{
+		HeadFetcher: &mock.ChainService{State: beaconState},
+		SyncChecker: &mockSync.Sync{IsSyncing: false},
+		BeaconDB:    db,
+		AttPool:     attestations.NewPool(),
+		P2p:         &mockp2p.MockBroadcaster{},
+	}
+
+	priv := bls.RandKey()
+	sig := priv.Sign([]byte{'B'}, 0)
+	pubKey := [48]byte{'B'}
+	req := &pb.AggregationRequest{CommitteeIndex: 1, SlotSignature: sig.Marshal(), PublicKey: pubKey[:]}
+	if err := aggregatorServer.BeaconDB.SaveValidatorIndex(ctx, pubKey, 100); err != nil {
+		t.Fatal(err)
+	}
+
+	if err := aggregatorServer.AttPool.SaveUnaggregatedAttestation(att0); err != nil {
+		t.Fatal(err)
+	}
+
+	if _, err := aggregatorServer.SubmitAggregateAndProof(ctx, req); err != nil {
+		t.Fatal(err)
+	}
+
+	aggregatedAtts := aggregatorServer.AttPool.AggregatedAttestation()
+	if len(aggregatedAtts) != 0 {
+		t.Errorf("Wanted aggregated attestation 0, got %d", len(aggregatedAtts))
+	}
+}
+
+func GenerateAtt(state *pbp2p.BeaconState, index uint64, privKeys []*bls.SecretKey) *ethpb.Attestation {
```

Comment:
Why exported? If this is used in various places can we put it in the testutil? There are some utils that generate blocks as well and would be nice to have these in one place

Reply:
You are right. It doesn't have to be exported